e2e_test/batch/string_concat_op.slt

```diff
@@ -0,0 +1,25 @@
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
+
+query T
+select 'a' || 'b' || 'c';
+----
+abc
+
+query T
+select '2' || NULL;
+----
+(empty)
```

Comment:
In Postgres, `concat('2', NULL)` gives `'2'` while `'2' || NULL` gives `NULL` since all operator applying to `NULL` should return `NULL`. This will fail since we simply forward it into `concat_ws` now.

Any good practice on this? cc @xiangjinwu :)

Reply:
I checked PostgreSQL implementation and they actually implemented separately without reusing:
* `concat` is a non-[strict](https://www.postgresql.org/docs/current/catalog-pg-proc.html#:~:text=not%20leak%2Dproof.-,proisstrict,-bool) variadic function that maps to [`text_concat`](https://github.com/postgres/postgres/blob/07d683b54af854098cc559d4b8640905f9efa0ea/src/backend/utils/adt/varlena.c#L5536).
  * `concat_ws` does share a common part with `concat`: `concat_internal`
* ` text || text` is a strict binary function that maps to [`textcat`](https://github.com/postgres/postgres/blob/07d683b54af854098cc559d4b8640905f9efa0ea/src/backend/utils/adt/varlena.c#L729).
  * `any || text` or `text || any` is [rewritten](https://github.com/postgres/postgres/blob/19408aae7fa2bf28866fb262b27f08405e71152e/src/include/catalog/pg_proc.dat#L5876) with explicit cast added.

Given their behavior on bool is also expected to be different (#3146), let's keep 2 implementations as well.
```
test=# select '1' || true;
 ?column? 
----------
 1true
(1 row)

test=# select concat('1', true);
 concat 
--------
 1t
(1 row)
```